src/codegen/riscv64/macro-assembler-riscv64.cc

```diff
@@ -444,20 +445,21 @@ void TurboAssembler::Sub64(Register rd, Register rs, const Operand& rt) {
          static_cast<int32_t>(
              -rt.immediate()));  // No subi instr, use addi(x, y, -imm).
   } else {
-    DCHECK(rs != t3);
     int li_count = InstrCountForLi64Bit(rt.immediate());
     int li_neg_count = InstrCountForLi64Bit(-rt.immediate());
     if (li_neg_count < li_count && !MustUseReg(rt.rmode())) {
       // Use load -imm and add when loading -imm generates one instruction.
       DCHECK(rt.immediate() != std::numeric_limits<int32_t>::min());
       UseScratchRegisterScope temps(this);
       Register scratch = temps.Acquire();
+      DCHECK(rs != scratch);
```

Comment:
I think this check is not necessary, because `temps.Acquire()` should only return an unused temp register, so it could not be equal to `rs`, right?